src/openforms/js/components/admin/form_design/logic/actions/Action.js

```diff
@@ -1,48 +1,54 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import {useIntl} from 'react-intl';
 
 import Select from '../../../forms/Select';
 import DeleteIcon from '../../../DeleteIcon';
-
 import {ACTION_TYPES} from '../constants';
 import DataPreview from '../DataPreview';
 import {ActionComponent} from './Actions';
 import {Action as ActionType} from './types';
+import ErrorList from '../../../forms/ErrorList';
 
 
-const Action = ({prefixText, action, onChange, onDelete}) => {
+const Action = ({prefixText, action, errors={}, onChange, onDelete}) => {
     const intl = useIntl();
+    const hasErrors = Object.entries(errors).length > 0;
 
     return (
         <div className="logic-action">
 
-            <div className="logic-action__actions">
-                <DeleteIcon
-                    onConfirm={onDelete}
-                    message={intl.formatMessage({
-                        description: 'Logic rule action deletion confirm message',
-                        defaultMessage: 'Are you sure you want to delete this action?',
-                    })}
-                />
-            </div>
+            <div className={`logic-action__row ${classNames({'logic-action__row--has-errors': hasErrors})}`}>
+
+                <div className="logic-action__actions">
+                    <DeleteIcon
+                        onConfirm={onDelete}
+                        message={intl.formatMessage({
+                            description: 'Logic rule action deletion confirm message',
+                            defaultMessage: 'Are you sure you want to delete this action?',
+                        })}
+                    />
+                </div>
 
-            <div className="logic-action__action">
-                <div className="dsl-editor">
-                    <div className="dsl-editor__node">{prefixText}</div>
+                <div className="logic-action__action">
+                    <div className="dsl-editor">
+                        <div className="dsl-editor__node">{prefixText}</div>
 
-                    <div className="dsl-editor__node">
-                        <Select
-                            name="action.type"
-                            choices={ACTION_TYPES}
-                            translateChoices
-                            allowBlank
-                            onChange={onChange}
-                            value={action.action.type}
-                        />
-                    </div>
+                        <div className={`dsl-editor__node ${classNames({'errors': !!errors.action?.type})}`}>
```

Comment:
following BEM the `errors` should be a modifier on the existing element:

```suggestion
                        <div className={`dsl-editor__node ${classNames({'dsl-editor__node--errors': !!errors.action?.type})}`}>
```